### ISSUE TITLE
lr_find does not return best learning rate (it returns nothing)

### ISSUE THREAD

**Issue by marcturpin91** [amaiya/ktrain#117]
The lr_find function states that it is supposed to return a float representing the estimate for best lr. It doesn't currently do that. It also doesnt seem to store the best lr anywhere in the lr_finder object, despite having two separate lists storing the individual losses and individual learning rates.

I also could not get the show_plot parameter to work, but that is less important and might be tied to some dependencies. I am running the latest version of TF (2.1.0).

`lr_find(self, start_lr=1e-7, lr_mult=1.01, max_epochs=None, 
                stop_factor=4, show_plot=False, verbose=1):
Returns:
            float:  Numerical estimate of best lr. `

**Comment by marcturpin91** [comment]
For now I changed the LRFinder object to have an initial variable of self.best_lr = 0.0

Then I updated it at the same location that the self.best_loss gets updated

`# record best loss
        if smoothed_loss < self.best_loss or self.batch_num==1:
            self.best_loss = smoothed_loss
            self.best_lr = lr`

**Comment by amaiya** [comment]
Hello @marcturpin91 

As of right now, the `lr_find` method does not return anything and is intended to be visually analyzed by user to select learning rate associated with decreasing/falling loss, as shown in the example notebooks.  The docstring for `lr_find` is a mistake that was never corrected.  Thanks for letting me know by posting an issue. 

Note that using the learning rate associated with best (or lowest) loss as you've done may not be ideal, as it may be an overestimate of the learning rate (learning rates associated with falling losses rather than the minimum loss tend to work better).

That said, there is an unreleased version of the LR-Finder that will infer the learning rate estimate from the plot using various strategies like minimum numerical gradient (which will be different from minimum/best loss) and highlight the plot accordingly along with printing or returning the estimate.  

I will mark this issue as enhancement for now and will try to release this option in next version.

Thanks again.

**Comment by marcturpin91** [comment]
Just want to give an update in regards to the lr_plot() not showing any plots. I believe it's because notebooks are 'interactive' and will show the plots automatically, but I am running locally in Pycharm. I fixed the issue by calling manually the show() function from pyplot, so it would display the plot that was generated.

```
from matplotlib import pyplot
learner.lr_plot()
pyplot.show()
```

**Comment by srulikbd** [comment]
@marcturpin91 the lr.plot() shows me an empty graph in PyCharm.
do you have any suggestions?

**Comment by amaiya** [comment]
@marcturpin91 I have [enabled interactive mode](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.ion.html) using matplotlib within *ktrain*.   I tested `learner.lr_find(show_plot=True)` and `learner.lr_plot()` outside of Jupyter and ipython using a vanilla Python console and the plots correctly appear now.  It should work now in PyCharm.  Thanks for reporting issue.

@srulikbd  If the default stop condition is being reached too early, you might get an empty graph.  You can change this by increasing `stop_factor` parameter to `lr_find`.  If that doesn't work, try reducing `batch_size`.

**Comment by frbuccoliero** [comment]
Hi there! I'm currently using trying to plot after lr_find() but the graph never appears. I tried also:
- adding show_plot=True inside the lr_find() function
- using @marcturpin91 's matplotlib method
Any tip? I'm currently on the latest ktrain with python3.9.8

**Comment by amaiya** [comment]
@frbuccoliero Are you able to reproduce this issue on Google Colab and send me a link to the notebook?

**Comment by frbuccoliero** [comment]
Ok I just copypasted the code on google colab and it gives me no error, it plots smoothly.
Well, since I did the lr_find() on my actual dataset may I just use the minimum value I got here and use it in my real code?

**Comment by amaiya** [comment]
@frbuccoliero Yes,  you can use the value from Google Colab as long as you used the same dataset (and model).  Please open a new issue if you have further questions/issues.  In the meantime,  I will try to see if I can reproduce the issue you're seeing locally.

**Comment by frbuccoliero** [comment]
Thanks, here the [colab](https://colab.research.google.com/drive/15StF5Ojb4pGFLGIdYdyfVf_7Wqn4Inrj?usp=sharing) just in case, but as you can see it's a very "getting started" code :D
